SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -359,7 +371,23 @@ bool FLocalDeploymentManager::TryStopLocalDeployment()
 	}
 
 	bStoppingDeployment = true;
-	RuntimeProcess->Stop();
+
+	// Prepare runtime process name
+	FString RuntimeProcessName = SpatialGDKServicesConstants::GetRuntimeExecutablePath(ActiveRuntimeVersion);
+	RuntimeProcessName = RuntimeProcessName.Replace(TEXT("/"),TEXT("\\"));
+
+	// Find runtime window
+	HWND RuntimeWindowHandle = FindWindowA(NULL, TCHAR_TO_ANSI(*RuntimeProcessName));
```

Comment:
hmmmmmmmmmmmmmmm
there must be a way throught the Unreal API rather than the windows one, no?

Reply:
also, this will probably break Mac builds

Reply:
probably....

Reply:
Hmmm, looking at FPlatformProcess::CloseProc, it seems to be the thing we want.

Reply:
yeah, I think we should try CloseProc if possible and see what happens.
It may be *too* soft...

Reply:
So the current flow makes use of `MonitoredProcess->Stop`, this eventually calls `TerminateProc` which 'should' according to the API signal the process: https://docs.microsoft.com/en-us/windows/win32/procthread/terminating-a-process. `CloseProc` eventually calls `CloseHandle`: https://docs.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-closehandle
I'm not sure if closing the handle does the same thing and sends a stop signal, worth a try but I'm under the impression `TerminateProc` will still be required too.

Reply:
Cool. I was using CloseHandle as one point. Issue with that was that it does not close the actual window. Will do some more experimenting though with the unreal API.

Reply:
Yeah FPlatformProcess::CloseProc does not actually close the process... :(

Reply:
The windows code we have hear seems to be correct but obviously only works on windows. Perhaps we should find a mac equivalent and wrap the two code blocks in #defines

Reply:
I guess we could always extend the Unreal process libs to also send signals for Windows/unix (the hwnd can be obtained from the process handle via EnumWindows and comparing process id I think)

Reply:
So are you suggesting we ignore mac? We might also be able to ignore linux as I do not think we support development on linux. Perhaps a line of actions is to fix it for windows and have it terminate for other platforms as the other platforms are kind of second class citizens.

Reply:
Does `CloseHandle` actually send the message to the runtime to stop? If so we can call `CloseHandle`/`CloseProc` followed by `TerminateProc`?

Reply:
Unfortunately this does not work :( Calling ClosProc then TerminateProc will not close the application. Perversely, simply calling TerminateProc on it's own will but obviously not gracefully.

Reply:
Hmmm, I suspect this may be because closing the handle means we lose the known pid. I wonder if we can store the pid and call a terminate on that after the close?

Reply:
> So are you suggesting we ignore mac?
I think Mac/Linux both share the same unix process core for this sort of stuff so should be easier there

Reply:
It does not seem to me that there is a single Unreal API call to get pid for windows, max and unix. Additionally Unreal doesn't seem to have any API calls for terminating a process for a given pid. I also have experimented with windows API calls CloseHandle and then TerminateHandle. This has the same effect as using the Unreal equivalent calls.

Reply:
Does the `runtime.exe` still exist? If so you could always make use of my trusty `SpatialCommandUtils::TryKillProcessWithName` ?

Reply:
Some success here but after more testing I am a little worried CloseProc isn't doing what we want it to. So calling SpatialCommandUtils::TryKillProcessWithName after CloseProc will actually terminate it. Brilliant. I am still not super convinced that the waiting logic is now valid. Also still concerned CloseProc does not actually do what we want it to do.